mesonbuild/modules/python.py

```diff
@@ -727,10 +727,23 @@ def find_installation(self, state: 'ModuleState', args: T.Tuple[T.Optional[str]]
         found_modules: T.List[str] = []
         missing_modules: T.List[str] = []
         if python.found() and want_modules:
+            # Python 3.8.x or later require add_dll_directory() to be called on Windows if
+            # the needed modules require external DLLs that are not bundled with the modules.
+            # Simplify things by calling add_dll_directory() on the paths in %PATH%
+            add_paths_cmd = ''
+            if hasattr(os, 'add_dll_directory'):
+                add_paths_cmds = []
+                paths = os.environ['PATH'].split(os.pathsep)
+                for path in paths:
+                    if path != '' and os.path.isdir(path):
+                        add_paths_cmds.append('os.add_dll_directory({!r})'.format(path))
```

Comment:
Since this is no longer a complex expression, this can become:

```suggestion
                        add_paths_cmds.append(f'os.add_dll_directory({path!r})')
```